remake.yml

```diff
@@ -15,17 +15,32 @@ targets:
   all:
     depends: 3_visualize/out/figure_1.png
 
-  download_files:
-    command: download_nwis_data(dl_dir = '1_fetch/out')
+  1_fetch/out/nwis_01427207_data.csv:
+    command: download_nwis_data(site_id = I("01427207"))
```

Comment:
This is not a sticking point, but more of an opportunity to point out one of the uses of `target_name`. 

You've already noted that it isn't scalable to have a _ton_ of these nwis files in the remakefile (which is true, and a good observation). But since our present solution requires some duplication and copy/paste, I would suggest making the places in each of those nearly identical targets where you have to edit/customize limited. So, this means if you can use the target_name to extract the `site_id`, it would be helpful, because then all of your `command: download_nwis_data(target_name)` could be identical

The original function helped you do that, like this

```r
  site_num <- basename(filepath) %>% 
    stringr::str_extract(pattern = "(?:[0-9]+)")
```

---

not a necessary change, just something to point out here